client/httplib/httplib.go

```diff
@@ -58,36 +58,44 @@ import (
 const contentTypeKey = "Content-Type"
 
 // it will be the last filter and execute request.Do
-var doRequestFilter = func(ctx context.Context, req *BeegoHTTPRequest) (*http.Response, error) {
-	return req.doRequest(ctx)
+var doRequestFilter = func(_ context.Context, req *BeegoHTTPRequest) (*http.Response, error) {
+	return req.doRequest()
 }
 
 // NewBeegoRequest returns *BeegoHttpRequest with specific method
 // TODO add error as return value
 // I think if we don't return error
 // users are hard to check whether we create Beego request successfully
 func NewBeegoRequest(rawurl, method string) *BeegoHTTPRequest {
-	var resp http.Response
-	u, err := url.Parse(rawurl)
+	req, err := NewBeegoRequestWithContext(context.Background(), rawurl, method)
+	if err != nil {
+		return &BeegoHTTPRequest{}
+	}
+
+	return req
+}
+
+// NewBeegoRequestWithContext returns a new BeegoHTTPRequest given a method, URL
+func NewBeegoRequestWithContext(ctx context.Context, rawurl, method string) (*BeegoHTTPRequest, error) {
```

Comment:
NewBeegoRequestWithCtx

Reply:
Because the BeegoRequest is the implementation of Builder pattern. So this function we should not return error. 
But I do agree with you that here we should tell users what happen.
So I think the possible solution is we add a new field named "err" in structure.
And when we call the terminative function like "DoRequest", we check the `if b.err != nil return b.err`